src/mosaic.mjs

```diff
@@ -87,14 +87,66 @@ function requestCachedMosaic512px(z, x, y) {
   return request;
 }
 
+async function bufferToPixels(buffer) {
+
+  if (!buffer || buffer.length === 0 || buffer === undefined) {
+    console.log('Input buffer is empty or undefined', !buffer, buffer.length);
+    return null;
+  }
+
+  const { data, info } = await sharp(buffer)
+    .raw()
+    .toBuffer({ resolveWithObject: true });
+
+  const pixels = [];
+  for (let i = 0; i < data.length; i += 4) {
+    pixels.push({
+      r: data[i],
+      g: data[i + 1],
+      b: data[i + 2],
+      a: data[i + 3]
+    });
+  }
+
+  return { pixels, width: info.width, height: info.height };
+}
+
+async function pixelsToBuffer(pixels, width, height) {
+  if (!pixels || pixels.length === 0 || pixels === undefined) {
+    console.log('Input pixels are empty or undefined');
+    return null;
+  }
+
+  const data = new Uint8Array(width * height * 4); // Keeping with RGBA format from the PNG
+  for (let i = 0; i < pixels.length; i++) {
+    const blackThreshold = 33;
+    if (pixels[i].r <= blackThreshold && pixels[i].g <= blackThreshold && pixels[i].b <= blackThreshold) {
+      data[i * 4 + 3] = 0;
+    }
+  }
+
+  // Use sharp to create a PNG buffer
+  const buffer = await sharp(data, { raw: { width, height, channels: 4 } })
+    .png()
+    .toBuffer();
+
+  return buffer;
+}
+
 async function cachedMosaic256px(z, x, y) {
   let tileBuffer = await cacheGetTile("__mosaic256__", z, x, y, "png");
+
   if (tileBuffer) {
```

Comment:
Why are you altering the tiles after stitching and after caching? The whole idea was that you replace sharp.composite with a better implementation, *during* the stitching: https://github.com/konturio/oam-mosaic-map/blob/7a142b040a61d43ae915ec8e5d6d40fcacfc93f0/src/mosaic.mjs#L228-L235